Mathlib/RingTheory/PowerSeries/CoeffMulMem.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2025 Jz Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jz Pan
+-/
+import Mathlib.RingTheory.Ideal.Operations
+import Mathlib.RingTheory.Ideal.BigOperators
+import Mathlib.RingTheory.PowerSeries.Basic
+
+/-!
+
+# Some results on the coefficients of multiplication of two power series
+
+## Main results
+
+- `PowerSeries.coeff_mul_mem_ideal_mul_ideal_of_coeff_mem_ideal`,
+  `PowerSeries.coeff_mul_mem_ideal_mul_ideal_of_coeff_mem_ideal'`:
+  if for all `i ≤ n` (resp. for all `i`), the `i`-th coefficients of power series `f` and `g` are
+  in ideals `I` and `J`, respectively, then for all `i ≤ n` (resp. for all `i`), the `i`-th
+  coefficients of `f * g` are in `I * J`.
+
+- `PowerSeries.coeff_mul_mem_ideal_of_coeff_right_mem_ideal`,
+  `PowerSeries.coeff_mul_mem_ideal_of_coeff_right_mem_ideal'`:
+  if for all `i ≤ n` (resp. for all `i`), the `i`-th coefficients of power series `g` are
+  in ideal `I`, then for all `i ≤ n` (resp. for all `i`), the `i`-th coefficients of `f * g` are
+  in `I`.
+
+- `PowerSeries.coeff_mul_mem_ideal_of_coeff_left_mem_ideal`,
+  `PowerSeries.coeff_mul_mem_ideal_of_coeff_left_mem_ideal'`:
+  if for all `i ≤ n` (resp. for all `i`), the `i`-th coefficients of power series `f` are
+  in ideal `I`, then for all `i ≤ n` (resp. for all `i`), the `i`-th coefficients of `f * g` are
+  in `I`.
+
+-/
+
+namespace PowerSeries
+
+variable {A : Type*} [Semiring A] {I J : Ideal A} {f g : A⟦X⟧} (n : ℕ)
+
+theorem coeff_mul_mem_ideal_mul_ideal_of_coeff_mem_ideal (hf : ∀ i ≤ n, coeff A i f ∈ I)
```

Comment:
Sorry for changing idea, but I see you have to use `:=` quite frequently. Feel free to make the variable explicit.

Reply:
No worries. In actual application it's OK to make `f` and `g` implicit.